pkg/gadgets/biotop/gadget.go

```diff
@@ -0,0 +1,202 @@
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	biotoptracer "github.com/kinvolk/inspektor-gadget/pkg/gadgets/biotop/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/biotop/types"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+)
+
+const (
+	maxRowsDefault  = 20
+	intervalDefault = 1
+	sortByDefault   = types.ALL
+)
+
+type Trace struct {
+	resolver gadgets.Resolver
+
+	started bool
+	tracer  *biotoptracer.Tracer
+}
+
+type TraceFactory struct {
+	gadgets.BaseFactory
+}
+
+func NewFactory() gadgets.TraceFactory {
+	return &TraceFactory{
+		BaseFactory: gadgets.BaseFactory{DeleteTrace: deleteTrace},
+	}
+}
+
+func (f *TraceFactory) Description() string {
+	t := `biotop shows command generating block I/O, with container details.
+
+The following parameters are supported:
+ - interval: Output interval, in seconds. (default %d)
+ - max_rows: Maximum rows to print. (default %d)
+ - sort: The field to sort the results by (%s). (default %s)`
+	return fmt.Sprintf(t, intervalDefault, maxRowsDefault,
+		strings.Join(types.SortBySlice, ","), sortByDefault)
+}
+
+func (f *TraceFactory) OutputModesSupported() map[string]struct{} {
+	return map[string]struct{}{
+		"Stream": {},
+	}
+}
+
+func deleteTrace(name string, t interface{}) {
+	trace := t.(*Trace)
+	if trace.tracer != nil {
+		trace.tracer.Stop()
+	}
+}
+
+func (f *TraceFactory) Operations() map[string]gadgets.TraceOperation {
+	n := func() interface{} {
+		return &Trace{
+			resolver: f.Resolver,
+		}
+	}
+
+	return map[string]gadgets.TraceOperation{
+		"start": {
+			Doc: "Start biotop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Start(trace)
+			},
+		},
+		"stop": {
+			Doc: "Stop biotop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Stop(trace)
+			},
+		},
+	}
+}
+
+func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
+	if t.started {
+		trace.Status.State = "Started"
+		return
+	}
+
+	traceName := gadgets.TraceName(trace.ObjectMeta.Namespace, trace.ObjectMeta.Name)
+
+	maxRows := maxRowsDefault
+	intervalSeconds := intervalDefault
+	sortBy := sortByDefault
+
+	if trace.Spec.Parameters != nil {
+		params := trace.Spec.Parameters
+		var err error
+
+		if val, ok := params["max_rows"]; ok {
+			maxRows, err = strconv.Atoi(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for ouput_rows", val)
+				return
+			}
+		}
+
+		if val, ok := params["interval"]; ok {
+			intervalSeconds, err = strconv.Atoi(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for interval", val)
+				return
+			}
+		}
+
+		if val, ok := params["sortby"]; ok {
+			sortBy, err = types.ParseSortBy(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for sortby", val)
+				return
+			}
+		}
```

Comment:
Same here, in the CLI it is defined as `sort_by`.

I expect these two parameters to not work with the CLI or what did I miss?